src/scimlfunctions.jl

```diff
@@ -4199,7 +4199,10 @@ IntervalNonlinearFunction(f::IntervalNonlinearFunction; kwargs...) = f
 struct NoAD <: AbstractADType end
 
 (f::OptimizationFunction)(args...) = f.f(args...)
-OptimizationFunction(args...; kwargs...) = OptimizationFunction{true}(args...; kwargs...)
+function OptimizationFunction(f, args...; kwargs...)
+    isinplace(f, 2, outofplace_param_number=2)
```

Comment:
This isn't needed though?

Reply:
The purpose of this is to ensure that the function has the correct number of arguments, which I believe we want to do.